hibernate-core/src/main/java/org/hibernate/loader/ast/internal/LoaderSelectBuilder.java

```diff
@@ -380,22 +381,20 @@ private void applyKeyRestriction(
 	}
 
 	private void applyFiltering(QuerySpec querySpec, TableGroup tableGroup, PluralAttributeMapping pluralAttributeMapping) {
-		if ( loadQueryInfluencers.hasEnabledFilters() ) {
-			final Joinable joinable = pluralAttributeMapping
-					.getCollectionDescriptor()
-					.getCollectionType()
-					.getAssociatedJoinable( creationContext.getSessionFactory() );
-			assert joinable instanceof AbstractCollectionPersister;
-			final String tableExpression = joinable.getTableName();
-			final String tableAlias = tableGroup.resolveTableReference( tableExpression ).getIdentificationVariable();
-			final Predicate filterPredicate = FilterHelper.createFilterPredicate(
-					loadQueryInfluencers,
-					joinable,
-					tableAlias
-			);
-			if ( filterPredicate != null ) {
-				querySpec.applyPredicate( filterPredicate );
-			}
+		final Joinable joinable = pluralAttributeMapping
+				.getCollectionDescriptor()
+				.getCollectionType()
+				.getAssociatedJoinable( creationContext.getSessionFactory() );
+		assert joinable instanceof AbstractCollectionPersister;
+		final String tableExpression = joinable.getTableName();
+		final String tableAlias = tableGroup.resolveTableReference( tableExpression ).getIdentificationVariable();
+		final Predicate filterPredicate = FilterHelper.createFilterPredicate(
+				loadQueryInfluencers,
+				joinable,
+				tableAlias
+		);
+		if ( filterPredicate != null ) {
+			querySpec.applyPredicate( filterPredicate );
```

Comment:
The above change is simply to get rid of `if ( loadQueryInfluencers.hasEnabledFilters() ) {` for it will add `@Where` and `@WhereJoinTable` filter content as well. This is the gist of this PR.